git/gogit/client.go

```diff
@@ -310,10 +308,14 @@ func (g *Client) Commit(info git.Commit, commitOpts ...repository.CommitOption)
 
 	for file := range status {
 		_, _ = wt.Add(file)
```

Comment:
not sure I understand why we do this here as well as after calling `g.writeFile` in the other for loop but changed as minimal as I could to get it working as previously

Reply:
My understanding is that it stages files that were changed at the `fs`, whether or not they were passed on via `options.Files`.

Please note that line `309` does the same loop as `status.IsClean()`:
```go
func (s Status) IsClean() bool {
	for _, status := range s {
		if status.Worktree != Unmodified || status.Staging != Unmodified {
			return false
		}
	}

	return true
}
```

So to avoid going through twice over the statuses, we could just add the `if status.Worktree != Unmodified || status.Staging != Unmodified` in `310` here. However, as I mentioned on the other comment, this does not seem to be the problem, as by default `go-git` should not allow "empty commits".